File: C++/Medium/WrongAnswer/splitting-a-string-into-descending-consecutive-values_7.cpp
class Solution {
public:
    bool splitString(string s) {
        
        return dfs(s, 0, 0, false);
    }
    bool dfs(string& s, int i, long long pre, bool flag) {
        if (i == s.size()) return flag;
        long long cur = 0;
        for (int j = i; j < s.size(); j++) {
            cur = cur * 10 + s[j] - '0';
            if (cur > pre) break;
            if (cur == pre - 1 && dfs(s, j + 1, cur, true)) return true;
        }
        return false;
    }
};

File: Rust/Medium/CompileError/splitting-a-string-into-descending-consecutive-values_7.rs
impl Solution {
    pub fn split_string(s: String) -> bool {
        
        let mut s = s.chars().collect::<Vec<char>>();
        let mut n = s.len();
        let mut res = false;
        let mut dfs = |i: usize, last: i32| {
            if res {
                return;
            }
            if i == n {
                res = true;
                return;
            }
            let mut x = 0;
            for j in i..n {
                x = x * 10 + (s[j] as i32 - '0' as i32);
                if x == last - 1 {
                    dfs(j + 1, x);
                }
            }
        };
        dfs(0, std::i32::MAX);
        res
    }
}